packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -544,38 +544,36 @@ class FlutterPlugin implements Plugin<Project> {
                 }
             }
             String variantBuildMode = buildModeFor(variant.buildType)
-            List<FlutterTask> compileTasks = targetPlatforms.collect { targetArch ->
-                String taskName = toCammelCase(["compile", FLUTTER_BUILD_PREFIX, variant.name, targetArch.replace('android-', '')])
-                project.tasks.create(name: taskName, type: FlutterTask) {
-                    flutterRoot this.flutterRoot
-                    flutterExecutable this.flutterExecutable
-                    buildMode variantBuildMode
-                    localEngine this.localEngine
-                    localEngineSrcPath this.localEngineSrcPath
-                    abi PLATFORM_ARCH_MAP[targetArch]
-                    targetPath target
-                    verbose isVerbose()
-                    fileSystemRoots fileSystemRootsValue
-                    fileSystemScheme fileSystemSchemeValue
-                    trackWidgetCreation trackWidgetCreationValue
-                    targetPlatform targetArch
-                    sourceDir project.file(project.flutter.source)
-                    intermediateDir project.file("${project.buildDir}/${AndroidProject.FD_INTERMEDIATES}/flutter/${variant.name}/${targetArch}")
-                    extraFrontEndOptions extraFrontEndOptionsValue
-                    extraGenSnapshotOptions extraGenSnapshotOptionsValue
-                }
+            String taskName = toCammelCase(["compile", FLUTTER_BUILD_PREFIX, variant.name])
```

Comment:
Instead of creating a task per ABI, I create a single task which builds all of the requested ABIs. This allows assemble to run concurrent gen_snapshots